modules/social_features/social_comment/social_comment.tokens.inc

```diff
@@ -316,7 +316,7 @@ function _social_comment_get_summary($text) {
   $summary = preg_replace('/\n|\r|\t/m', ' ', $summary);
   $max_length = 280;
 
-  if (mb_strlen($summary) > $max_length) {
+  if (is_string($summary) && (mb_strlen($summary) > $max_length)) {
```

Comment:
This is a place where we could also use something like:
```
  $summary = preg_replace('/\n|\r|\t/m', ' ', $summary);
  assert($summary !== NULL, "Invalid regular expression in preg_replace call");
  $max_length = 280;

  if ((mb_strlen($summary) > $max_length)) {
  ```
 
I believe PHPStan is smart enough to know for this function that string in is string out and array in is array out.

This helps us more explicitly handle the error case rather than only looking for the happy path. In case the return type of preg_replace changes in the future (or $summary becomes an array) we'd get a new error from PHPStan warning us that our code is no longer working.

By using `is_string` we will not have PHPStan help us in those scenarios.

Reply:
Done